File: backjoon/dfs_bfs/2606_바이러스.py
from collections import deque

n = int(input())
graph = [[] for _ in range(n+1)]
m = int(input())
visited = [False]*(n+1)
count = 0
def bfs(graph, start, visited, count):
    queue = deque([start])
    visited[start] = True
    while queue:
        v = queue.popleft()
        count += 1
        for i in graph[v]:
            if not visited[i]:
                queue.append(i)
                visited[i] = True
    return count

# 양 방향으로 추가 필요
for _ in range(m):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)

print(bfs(graph, 1, visited, count)-1)

File: backjoon/dfs_bfs/2606_바이러스.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});
const input = [];

const solution = (input) => {
  let answer = 0;
  const [node, edge, ...rest] = input;
  const graph = new Array(+node + 1).fill(null).map((v) => new Array());
  let queue = [];
  const visited = new Array(+node + 1).fill(null).map((v) => false);
  for (let i = 0; i < edge; i++) {
    const [from, to] = rest[i].split(" ");
    graph[from].push(to);
    graph[to].push(from);
  }
  visited[1] = true;
  queue.push(1);
  while (queue.length > 0) {
    const shiftNode = queue.shift();
    graph[shiftNode]?.forEach((v) => {
      if (!visited[+v]) {
        answer += 1;
        visited[+v] = true;
        queue.push(v);
      }
    });
  }
  return answer;
};

rl.on("line", (answer) => {
  input.push(answer.trim());
}).on("close", () => {
  console.log(solution(input));
});
